codecov_cli/plugins/__init__.py

```diff
@@ -30,18 +30,28 @@ def select_preparation_plugins(cli_config: typing.Dict, plugin_names: typing.Lis
 
 def _load_plugin_from_yaml(plugin_dict: typing.Dict):
     try:
-        class_obj = import_module(plugin_dict["path"])
+        module_obj = import_module(plugin_dict["module"])
+        class_obj = getattr(module_obj, plugin_dict["class"])
     except ModuleNotFoundError:
         click.secho(
-            f"Unable to dynamically load plugin on path {plugin_dict['path']}",
+            f"Unable to dynamically load module {plugin_dict['module']}",
+            err=True,
+        )
+        return NoopPlugin()
+    except AttributeError:
+        click.secho(
+            f"Unable to dynamically load class {plugin_dict['class']} from module {plugin_dict['module']}",
             err=True,
         )
         return NoopPlugin()
     try:
-        return class_obj(**plugin_dict["params"])
+        if "params" not in plugin_dict or plugin_dict["params"] is None:
```

Comment:
[nit] what you wrote is perfectly fine. This is simply a suggestion for being more concise. You could do `params = plugin_dict.get("params", None)` and change the considitonal to `if params:` (and reuse `params` in the `else` too)